src/boot/efi/xbootldr.c

```diff
@@ -149,11 +133,18 @@ static EFI_STATUS try_gpt(
                 if (end < start) /* Bogus? */
                         continue;
 
-                ret_hd->PartitionNumber = i + 1;
-                ret_hd->PartitionStart = start;
-                ret_hd->PartitionSize = end - start + 1;
-                ret_hd->MBRType = MBR_TYPE_EFI_PARTITION_TABLE_HEADER;
-                ret_hd->SignatureType = SIGNATURE_TYPE_GUID;
+                *ret_hd = (HARDDRIVE_DEVICE_PATH) {
+                        .Header = (EFI_DEVICE_PATH) {
+                                .Type = MEDIA_DEVICE_PATH,
+                                .SubType = MEDIA_HARDDRIVE_DP,
+                                .Length = { 42 /* sizeof(HARDDRIVE_DEVICE_PATH) - padding */, 0 },
```

Comment:
why not specify the field name where to assing the 42 and the 0 to as well?

Reply:
The spec has Length as a `uint16_t`, but both gnu-efi and edk2 define it as a `uint8_t[2]` to force `EFI_DEVICE_PATH` to have alignment of 1. (And this is what `DevicePathNodeLength` is for.

Reply:
The definition is never gonna change, so I fail to see the reason. If the definition is ever gonna change, it will break code anyway (not just here but in initrd.c too).